### ISSUE TITLE
Search filter broken for tags with underscore

### ISSUE THREAD

**Issue by rmcrackan** [rmcrackan/Libation#655]
Technically, there's no need for this tag to exist in order to see the error. It will be needed for testing though.

**Describe the bug**
Search filter throws error when a tag-search includes underscores.

**To Reproduce**

1. Include this in a book's tags `free_audible_originals`
2. Filter by `-[free_audible_originals]`
3. Error

```
Bad filter string:

Cannot parse '-[free_audible_originals]': Encountered " "]" "] "" at line 1, column 24.
Was expecting one of:
    "TO" ...
    <RANGEIN_QUOTED> ...
    <RANGEIN_GOOP> ...
```

**Expected behavior**
Any book containing the tag `free_audible_originals` should not be included in grid results when search filter is `-[free_audible_originals]`. No error should be thrown.

Ideally, unit tests should also test for cases such as `[foo_bar]` , `-[foo_bar]` , `whatever [foo_bar]`

**Comment by Mbucari** [comment]
Lucene's `TokenStream` is splitting these strings into multiple tokens at the underscore, screwing up the parser.

I think this may be a bug in Lucene. And underscore followed by a letter will cause the string to be split into two tokens, but an underscore followed by a number (e.g. `foo_1bar`) will parse as a single token.

I changed lucene to tread underscores as alphabetic letters and that fixes this bug and doesn't break any other tests, but I'm not sure if it will break anything else. Can you think of any other ways underscores may be used that would be a problem if they were treated as letters?

**Comment by rmcrackan** [comment]
I've been using underscores in my default filter for years; they've been treated like a work boundary. Any of these searches would match `free_audible_originals`:

* `[free]`
* `[audible]`
* `[originals]`
* `[free_audible]`
* `[audible_originals]`
* `[free_audible_originals]`

**Comment by Mbucari** [comment]
@rmcrackan Yeah, that worked because you were using a regex to re-format tag query instead of the tokenizer. I still think splitting a token at an underscore followed by a letter is a bug. I looked at an older Libation release before the search refactor and I noticed that the query parser would turn `[foo_bar]` into `tags:"foo bar"`, removing the underscore. That string should not be split, but it is because of the aforementioned bug.

There are two ways we can fix this:
1. Change the lucene implementation to treat an underscore like a letter
2. Work around the bug in Libation.
